pkg/operator/ceph/cluster/osd/status.go

```diff
@@ -125,313 +156,215 @@ func parseOrchestrationStatus(data map[string]string) *OrchestrationStatus {
-	// be used to determine if we have gone past our timeout without making progress.
-	timeoutTimer := time.NewTimer(time.Minute)
-	shouldResetTimeoutTimer := true // set true when timeout timer should be reset
-	currentTimeoutMinutes := 0      // track how many minutes have gone by without progress
+	// tick after a short time of waiting for new OSD provision status configmaps to change state
+	// in order to allow opportunistic deployment updates while we wait
+	updateTicker := time.NewTicker(osdOpportunisticUpdateDuration)
+	defer updateTicker.Stop()
 
+	watchErrMsg := "failed during watch of OSD provisioning status ConfigMaps"
 	for {
-		// Check whether we need to cancel the orchestration
-		if err := controller.CheckForCancelledOrchestration(c.context); err != nil {
-			config.addError("%s", err.Error())
-			return false
+		if updateConfig.doneUpdating() && createConfig.doneCreating() {
+			break // loop
 		}
 
-		opts := metav1.ListOptions{
-			LabelSelector: selector,
-			Watch:         true,
-			// start watching for changes on the orchestration status map
-			ResourceVersion: statuses.ResourceVersion,
+		// reset the update ticker (and drain the channel if necessary) to make sure we always
+		// wait a little bit for an OSD prepare result before opportunistically updating deployments
+		updateTicker.Reset(osdOpportunisticUpdateDuration)
+		if len(updateTicker.C) > 0 {
+			<-updateTicker.C
 		}
-		logger.Infof("%d/%d node(s) completed osd provisioning, resource version %v", (originalNodes - remainingNodes.Count()), originalNodes, opts.ResourceVersion)
 
-		w, err := c.context.Clientset.CoreV1().ConfigMaps(c.clusterInfo.Namespace).Watch(ctx, opts)
-		if err != nil {
-			logger.Warningf("failed to start watch on osd status, trying again. %v", err)
-			time.Sleep(5 * time.Second)
-			continue
-		}
-		defer w.Stop()
-
-	ResultLoop:
-		for {
-			// Reset the opportunity timer every time we start this loop. If the timer fires before
-			// there is a ConfigMap update (new OSD created), we opportunistically handle an OSD
-			// update and then return to the top of this loop. If the ConfigMap watcher has a result
-			// before the timeout, a new OSD will be created. The opportunity timer might fire while
-			// creation is happening. When the OSD is done being created, we return to the top of
-			// this loop and reset the timeout here to make sure a previous timer firing doesn't
-			// take update precedence away from any ConfigMap updates (new OSD) that need handled.
-			if !opportunityTimer.Stop() {
-				// the timer already fired
-				if len(opportunityTimer.C) > 0 {
-					// the timer fired, but the result channel needs drained before Reset below
-					<-opportunityTimer.C
-				}
+		select {
+		case event, ok := <-watcher.ResultChan():
+			if !ok {
+				logger.Infof("restarting watcher for OSD provisioning status ConfigMaps. the watcher closed the channel")
```

Comment:
For users, I doubt it has much use. But since it doesn't happen much in the real world, I think it would be nice to leave it as `Info` at least until we are certain that this behaves as we need it to for the ETCD cache clears. I guess I don't think it's hurting as-is, and there is potential benefit.